storage/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteLogManager.java

```diff
@@ -973,6 +973,9 @@ public void copyLogSegmentsToRemote(UnifiedLog log) throws InterruptedException
                             segmentIdsBeingCopied.add(segmentId);
                             try {
                                 copyLogSegment(log, candidateLogSegment.logSegment, segmentId, candidateLogSegment.nextSegmentOffset);
+                            } catch (RemoteStorageException e) {
```

Comment:
the `copyLogSegment` can throw InterruptedException, ExecutionException, RemoteStorageException, IOException, and CustomMetadataSizeLimitExceededException. 

can we move the `recordLagStats` to the outer catch-block instead to handle CustomMetadataSizeLimitExceededException / InterruptedException / RetriableException / Exception?